Include/pybuffer.h

```diff
@@ -17,6 +17,9 @@ extern "C" {
  *
  */
 
+// Forward declaration
+typedef struct _object PyObject;
```

Comment:
That's ugly. I had a forward declaration in my first PR. @encukou asked me to remove it.

Reply:
How is it "ugly"? Could you be more specific?

There are many similar code used in Python header files to fix inter-dependency issues. See for example Include/pystate.h.

Reply:
I didn't realize the reason it was there. It's ugly, but it might be a good way to solve the issue.

Reply:
I elaborated the comment to explain why this "Forward declaration" is needed. Is it better?

Reply:
Why not forward-define `Py_buffer` in object.h instead? That's a smaller change anyway, and removes the ordering requirement completely, rather than simply forcing it to happen in a certain way.

Reply:
I see Py_buffer as a C structure which is unrelated to Python objects. It's like a low-level thing "under" PyObject. In practice, it contains a ``PyObject*`` pointer and pybuffer.h contains other function definitions which use ``PyObject*``.

I don't think that the order matters much, both orders work.

Reply:
> Why not forward-define Py_buffer in object.h instead?

It's not possible: the Py_buffer struct has no name on purpose, see @tiran's comment.